### PR TITLE
766 support raw api specification linting

### LINKED ISSUE #766: Support for raw API specifications linting
Currently, the `/api-violations` endpoint supports two ways of supplying the API specification: as a JSON and as a link. We should aim for having the raw presentation of the specification. This will allow us to check even more things (e.g. formatting) and provide better violation information for the user (e.g. line numbers) in the future.

**Acceptance criteria**
- [x] Zally API specification is extended to support the raw input as a string.
- [x] Controller prefers to lint the raw specification.
- [x] Zally CLI sends API specification documents as raw strings.

**Notes**
All the tooling should prefer to lint raw API specifications.

### PR DESCRIPTION
**Pull request by maxim-tschumak**
Closes #766

### CODE REVIEW

File: server/src/main/resources/api/zally-api.yaml [hunk 1]
```diff
@@ -195,11 +195,16 @@ paths:
...
     type: object
-    description: Request containing a swagger definition file
+    description: |
+      Request containing a swagger definition file. One of the following fields must be provided:
+      api_definition, api_definition_raw, or api_definition_url.
     properties:
       api_definition:
         type: object
         description: Specification object in OpenAPI format
+      api_definition_raw:
```
Comment (tkrop) [inline server/src/main/resources/api/zally-api.yaml]: Didn't we want to use consistent specification?

Anyhow, lets rename this to `_string`. Raw is a very unspecific term here.

Comment (maxim-tschumak) [reply server/src/main/resources/api/zally-api.yaml]: > Didn't we want to use consistent specification?

Yes, but having `api_defintition` and `api_specification_(raw|string)` in the same endpoint is even worse...

> Anyhow, lets rename this to _string. Raw is a very unspecific term here.

OK

**Comment by maxim-tschumak** [comment]
@tkrop I've renamed the new field and rebased the branch. Please have another look.

**Comment by ddufourboivin-zalando** [comment]
👍

**Comment by maxim-tschumak** [comment]
:+1:

File: server/src/main/resources/api/zally-api.yaml [hunk 2]
```diff
@@ -195,11 +195,16 @@ paths:
...
     type: object
-    description: Request containing a swagger definition file
+    description: |
+      Request containing a swagger definition file. One of the following fields must be provided:
+      api_definition, api_definition_string, or api_definition_url.
     properties:
       api_definition:
         type: object
         description: Specification object in OpenAPI format
+      api_definition_string:
```
Comment (roxspring) [inline server/src/main/resources/api/zally-api.yaml]: Interesting. I'd assumed that #766 would mean that I could simply POST an application/yaml (or application/json) body and have it validated and radically simplify the calling API. Was that considered and discarded? Curious to understand why if so!

**Pull request merged**